src/renderer/src/views/MigratingProject/index.tsx

```diff
@@ -0,0 +1,170 @@
+import { Button, Container, LinearProgress, Typography, useTheme } from '@mui/material'
+import CheckIcon from '@mui/icons-material/CheckCircleRounded'
+import { Column, Row } from '@renderer/components/LayoutComponents'
+import { Logo, Mapeo5Logo } from '@renderer/components/Logo'
+import { DefaultLayout } from '@renderer/layouts/default'
+import { useEffect, useState } from 'react'
+import { defineMessages, useIntl } from 'react-intl'
+import { useLocation } from 'react-router-dom'
+
+const INTERVAL_DURATION = 250
+
+type LocationState = { observations: number; media: number }
+
+export const MigratingProjectView = () => {
+  const theme = useTheme()
+  const { state } = useLocation()
+  const locationState = state as LocationState
+  const intl = useIntl()
+  const observationsTotal = locationState.observations
+  const mediaTotal = locationState.media
+  const total = observationsTotal + mediaTotal
+  const [observationsMigrated, setObservationsMigrated] = useState(0)
+  const [mediaMigrated, setMediaMigrated] = useState(0)
+  let interval: NodeJS.Timer
+
+  const progress = ((mediaMigrated + observationsMigrated) / total) * 100
+
+  useEffect(() => {
+    interval = setInterval(updateProgress, INTERVAL_DURATION)
+
+    return () => {
+      clearInterval(interval)
+    }
+  }, [])
```

Comment:
TBH, im not 100% sure of the dogma behind this, but in this case we can `useCallback` `updateProgress` and add it as a dependency to this `useEffect`. This will cause it to be a truely functional `useEffect`, but will still only run once.